webserver/cWebem.cpp

```diff
@@ -1157,6 +1157,115 @@ namespace http {
 			m_session_clean_timer.async_wait([this](auto &&) { CleanSessions(); });
 		}
 
+		bool cWebem::findRealHostBehindProxies(const request &req, std::string &realhost)
+		{
+			// Checking for 3 possible headers (in order of handling)
+			// "Forwarded"	RFC7239  (https://www.rfc-editor.org/rfc/rfc7239)
+			// "X-Forwarded-For" The defacto standard header used by many web/proxy servers (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For)
+			// "X-Real-IP"	The (old) default header used by NGINX  (http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header)
+			//
+			// These headers can occur multiple times, so need to be 'squashed' together
+			// And a single line can contain multiple (comma separated) values in order
+
+			bool bStatus = true;
+			std::vector<std::string> headers;
+			std::vector<std::string> hosts;
+
+			if (sumProxyHeader("Forwarded", req, headers))
+			{	// We found one or more Forwarded headers that need to be processed into a list of Hosts
+				if (!parseForwardedProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+			else if (sumProxyHeader("X-Forwarded-For", req, headers))
+			{	// We found one or more X-Forwarded-For headers that need to be processed into a list of Hosts
+				if (!parseProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+			else if (sumProxyHeader("X-Real-IP", req, headers))
+			{	// We found one or more X-Real-IP headers that need to be processed into a list of Hosts
+				if (!parseProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+
+			if (hosts.size() >= 1)
+			{
+				realhost = hosts[0];	// Even if we found a chain of hosts, we always use the first (= origin)
```

Comment:
instead of a bStatus boolean we could return true here or else return false at the bottom of this function?

Reply:
the True/False is about if we were able to process the headers correctly yes or no. That is independent of the fact if we found alternative hosts yes or no. Checking if an actual alternative host is found and what to do with it, is left up to the caller.

Personally I prefer to exit a function (`return`) at 1 (or as least as possible) location in the code so it is clear where and what the exit does. Improves readability and future maintenance (IMO).